lib/chai/core/assertions.js

```diff
@@ -1351,11 +1359,13 @@ module.exports = function (chai, _) {
 
   function assertKeys (keys) {
     var obj = flag(this, 'object')
+      , isDeep = flag(this, 'deep')
       , str
       , ok = true
       , mixedArgsMsg = 'when testing keys against an object or an array you must give a single Array|Object|String argument or multiple String arguments';
 
     if (_.type(obj) === 'map' || _.type(obj) === 'set') {
+      str = isDeep ? 'deeply ' : '';
```

Comment:
You are right!
I'm going to use a different variable to store that String, thanks 😄